Assets/Scripts/UI/HUDManager.cs

```diff
@@ -0,0 +1,68 @@
+using FMS.TapperRedone.Inventory;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class HUDManager : MonoBehaviour
+{
+
+    [SerializeField] private Image drinkHUDImage;        // Drag your DrinkHUDImage here in the inspector
+    [SerializeField] private Sprite defaultSprite;       // Sprite to show when no drink is held
+
+    private PlayerInventory playerInventory;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        // Assuming PlayerInventory is attached to the player
+        playerInventory = FindObjectOfType<PlayerInventory>();
+
+       // Ensures activation of HUD
+        if (playerInventory != null)
+        {
+            playerInventory.OnDrinkChanged += UpdateDrinkHUD; 
+        }
+
+        // Initialize HUD
+        UpdateDrinkHUD();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    private void UpdateDrinkHUD()
+    {
+        if (playerInventory.HasDrink())
+        {
+            // Get the first drink in the inventory
+            Item drink = playerInventory.itemsList.Find(item => item.IsDrink);
+            
+            //update  drink image
```

Comment:
You don't need a comment line for each code line. The code lines are pretty self-sufficient/self-descriptive as it is.

Reply:
true that i overcommented, is it a big problem for now?

Reply:
Better to get good habits early, right